parquet-column/src/main/java/parquet/filter2/recordlevel/FilteringPrimitiveConverter.java

```diff
@@ -34,29 +35,38 @@
 public class FilteringPrimitiveConverter extends PrimitiveConverter {
   private final PrimitiveConverter delegate;
   private final ValueInspector[] valueInspectors;
+  private Binding binding;
 
   public FilteringPrimitiveConverter(PrimitiveConverter delegate, ValueInspector[] valueInspectors) {
     this.delegate = checkNotNull(delegate, "delegate");
     this.valueInspectors = checkNotNull(valueInspectors, "valueInspectors");
   }
 
-  // TODO: this works, but
-  // TODO: essentially turns off the benefits of dictionary support
-  // TODO: even if the underlying delegate supports it.
-  // TODO: we should support it here. (https://issues.apache.org/jira/browse/PARQUET-36)
   @Override
   public boolean hasDictionarySupport() {
-    return false;
+    return true;
```

Comment:
Shouldn't this return `delegate.hasDictionarySupport()`?

Reply:
Hi

I'm not sure. It's 4 years since this PR, so it's pretty stale now.

I could try to resurrect it but I don't know if there is much interest from the project.

Mick